### ISSUE TITLE
Cannot use Java code generated from C#-written libraries in AndroidJavaSource files

### ISSUE THREAD

**Issue by aliyailina** [dotnet/android#9374]
### Android framework version

net8.0-android

### Affected platform version

.NET 8.0.303

### Description

Issue inspired by #9334 recommended workaround. I tried to override specific method in Application (precisely, MvxAndroidApplication) class inheritor, which was not possible due to .NET bug, so I asked for workaround. The workaround was to use generated Java class of MvxAndroidApplication, which was impossible due to `package crc6466d8e86b1ec8bfa8 does not exist` error. I tried it with full bin/obj delete + clean + rebuild process, but then found that error happens and Java's MvxAndroidApplication is not even generated, so then tried to first successfully build project without my custom AndroidJavaSource file included, then include and build again. Nothing works.

Also tried wit CommunityToolit.Maui's `MauiApplication` class, same error, so problem (I suppose) is not in MvvmCross.

Looks pretty similar #8191.

Here is [sample](https://github.com/user-attachments/files/17262939/JcwUsageSample.zip) and [build log](https://github.com/user-attachments/files/17262929/JetBrainsLog.ReSharperBuild_2024_10_04_19_20_42_0130.log)

### Steps to Reproduce

Create new clear project, add MvvmCross library via NuGet, create .java file with this content:
```
package jcwusagesample;
public class BaseApplication extends crc6466d8e86b1ec8bfa8.MvxAndroidApplication {
 }
```

and try to build.

Also you can check sample attached in Description section.

### Did you find any workaround?

Nope, but would be grateful for any workaround.

**Comment by dellis1972** [comment]
Your problem here is that the MvxAndroidApplication exists in MvvmCross.dll. So it is not a java type it is a C# type, so the type does not exist in any .jar file which is referenced by the project. 

One of the limitations mentioned in the docs is that you can only use types which exist in .jar files (https://github.com/dotnet/android/blob/4e4bc088400aa7f42721f6e7e888f6a778b1d10f/Documentation/guides/BindingJavaSource.md?plain=1#L88). 

Unfortuately any "java" code that might be generated won't be generated until after the Compile step of the main application has already been run. This is because `AndroidJavaSource` compilation happens before the Compile step, since its principal purpose is to bind java code. In this case though no java code exists.

**Comment by aliyailina** [comment]
@dellis1972 Yep, that is what I see either. I filed and issue because that was recommended by @jonpryor, so was expected.to work.

So C# classes referenced from .dll's are not supposed to be used in AndroidJavaSource files?

**Comment by dellis1972** [comment]
So if you want to `Bind` the Java source then no, you cannot use the C# types because there is no java class files available at the time that runs. 

If you don't want to bind it, then you probably can as the compilation of non-bindable .java files happens after _GenerateJAvaStubs. 

So if you add 

```
    <ItemGroup>
      <AndroidJavaSource Update="BaseApplication.java" Bind="false" />
    </ItemGroup>
```
It will compile. You problem will be the java code you just compiled won't be available in C#.

**Comment by aliyailina** [comment]
@dellis1972 Thanks, that is what I thought. May I ask you then help with any workaround for #9334 (if there is any)?
